int32 num_jit
int32 degree
---
bool complete